bitnami/kafka/templates/_helpers.tpl

```diff
@@ -385,6 +386,16 @@ kafka: rbac.create
 {{- end -}}
 {{- end -}}
 
+{{/* Validate values of Kafka - number of replicas must be the same as loadBalancerNames list */}}
+{{- define "kafka.validateValues.externalAccessServiceLoadBalancerNames" -}}
+{{- $replicaCount := int .Values.replicaCount }}
+{{- $loadBalancerNameListLength := len .Values.externalAccess.service.loadBalancerNames -}}
+{{- if and .Values.externalAccess.enabled (not .Values.externalAccess.autoDiscovery.enabled) (gt $loadBalancerNameListLength 0) (not (eq $replicaCount $loadBalancerNameListLength)) (eq .Values.externalAccess.service.type "LoadBalancer") }}
+kafka: externalAccess.service.loadBalancerNames
+    Number of replicas and loadBalancerNames array length must be the same. Currently: replicaCount = {{ $replicaCount }} and loadBalancerNames = {{ $loadBalancerNameListLength }}
+{{- end -}}
```

Comment:
This is not considering the possibility that a user sets `externalAccess.service.loadBalancerIPs ` instead. I mean, this will block the release even a user sets a list of IPs which length is equal to the number or replicas.

Reply:
`(gt $loadBalancerNameListLength 0)` avoids that scenario. If `externalAccess.service.loadBalancerNames` is not set  It won't be validated